foundation/plugins/org.jboss.tools.foundation.core/src/org/jboss/tools/foundation/core/credentials/internal/CredentialDomain.java

```diff
@@ -59,8 +65,20 @@ public CredentialDomain(Preferences pref) throws BackingStoreException {
 				credentials.put(users[i], NOT_LOADED_PASSWORD);
 			}
 		}
-		if( defaultUsername == null && credentials.size() > 0 ) {
-			defaultUsername = credentials.keySet().iterator().next();
+		
+		String promptedUserList = pref.get(PROPERTY_PROMPTED_USER_LIST, (String)null);
+		promptedCredentials = new ArrayList<String>();
+		if( !promptedUserList.isEmpty()) {
+			String[] users = (usersList == null ? new String[0] : promptedUserList.split("\n"));
+			promptedCredentials.addAll(Arrays.asList(users));
+		}
+		
+		if( defaultUsername == null || !userExists(defaultUsername)) {
+			// The default name doesn't exist, so we need another. 
+			String[] users = getUsernames();
+			if( users.length > 0) {
```

Comment:
why do we pick an arbitrary username? It'd be ok if user.length = 0, else better let the user explicitly choose one